src/passes/StringLowering.cpp

```diff
@@ -267,6 +267,44 @@ struct StringLowering : public StringGathering {
   Type nnExt = Type(HeapType::ext, NonNullable);
 
   void updateTypes(Module* module) {
+    // TypeMapper will not handle public types, but we do want to modify them as
+    // well: we are modifying the public ABI here. We can't simply tell
+    // TypeMapper to consider them private, as then they'd end up in the new big
+    // rec group with the private types (and as they are public, that would make
```

Comment:
Isn't this going to be an issue any time we tell `TypeMapper` to consider some public type as if it were private? Is it ever actually safe to treat a public type as private?

Reply:
I think #6630 is an example of that. We are certain the type can be modified there, despite it being public. We leave the public type alone but replace references to that type with modified ones.